[CI] IndexFoldersDeletionListenerIT testListenersInvokedWhenIndexIsDangling failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #5066 / windows-2019_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/5xbnb3t7b656c)
- [elasticsearch-periodic #5046 / release-tests](https://gradle-enterprise.elastic.co/s/6nycougbbcwis)

**Reproduction Line:**
```
gradlew ':server:internalClusterTest' --tests "org.elasticsearch.plugins.IndexFoldersDeletionListenerIT.testListenersInvokedWhenIndexIsDangling" -Dtests.seed=792CA2393BA6AAE3 -Dtests.locale=sat-IN -Dtests.timezone=Asia/Kuala_Lumpur -Druntime.java=22
```

**Applicable branches:**
8.15

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.plugins.IndexFoldersDeletionListenerIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testListenersInvokedWhenIndexIsDangling),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: java.lang.AssertionError: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 0
```

**Issue Reasons:**
- [8.15] 2 failures in test testListenersInvokedWhenIndexIsDangling (0.9% fail rate in 233 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/server/src/internalClusterTest/java/org/elasticsearch/plugins/IndexFoldersDeletionListenerIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/plugins/IndexFoldersDeletionListenerIT.java
@@ -350,7 +350,7 @@ public class IndexFoldersDeletionListenerIT extends ESIntegTestCase {
 
                 @Override
                 public void beforeShardFoldersDeleted(ShardId shardId, IndexSettings indexSettings, Path[] shardPaths) {
-                    deletedShards.computeIfAbsent(shardId.getIndex(), i -> new ArrayList<>()).add(shardId);
+                    deletedShards.computeIfAbsent(shardId.getIndex(), i -> Collections.synchronizedList(new ArrayList<>())).add(shardId);
                 }
             });
         }

